test/realm_logger_test.dart

```diff
@@ -0,0 +1,337 @@
+    Map<String, Map<Level, int>> logMessages = await Isolate.run(() async {
+      Map<String, Map<Level, int>> results = {};
+      Realm.logger.level = RealmLogLevel.error;
+
+      for (var i = 0; i < 2; i++) {
+        String attemptNum = "Attempt $i";
+        String generatedMessage = generateRandomString(15);
+        final allMessages = await _attachToLoggerBeforeAction("root isolate", () => _simulateLogs(generatedMessage));
+        final matchedMessages = _matchedMessagesCountPerLevel(allMessages, generatedMessage);
+        results.addEntries([MapEntry(attemptNum, matchedMessages)]);
+        Realm.logger.level = RealmLogLevel.off;
+      }
+      return results;
+    });
+    _expectErrorLogs(logMessages, "Attempt 0");
+
+    //After setting the log level to "Off" the logger won't receive the error
+    _expectNoLogs(logMessages, "Attempt 1");
+  });
+
+  test('Realm.logger logs messages from all the isolates', () async {
+    String generatedName = generateRandomString(15);
+    String rootIsolateName = "Root", isolate1Name = "Isolate1", isolate2Name = "Isolate2", rootIsolateAfterKillIsolatesName = "AfterKillIsolates";
+
+    Map<String, int> results = await Isolate.run(() async {
+      Map<String, int> results = {};
+      action() => _simulateLogs(generatedName);
+
+      // Root Isolate  with level Error. Attach to a new logger.
+      final rootIsolate = _attachToLoggerBeforeAction(rootIsolateName, action, setNewLogger: true, logLevel: RealmLogLevel.error);
+
+      // Isolate 1 with level Error. Attach to the existing logger.
+      ReceivePort isolate1ReceivePort = ReceivePort();
+      final isolate1 = Isolate.spawn((SendPort sendPort) async {
+        final messages = await _attachToLoggerBeforeAction(isolate1Name, action, logLevel: RealmLogLevel.error);
+        final matches = _matchedMessagesCountPerLevel(messages, generatedName);
+        sendPort.send(matches[RealmLogLevel.error]!);
+      }, isolate1ReceivePort.sendPort);
+
+      // Isolate 2 with level Error. Attach to a new logger.
+      ReceivePort isolate2ReceivePort = ReceivePort();
+      final isolate2 = Isolate.spawn((SendPort sendPort) async {
+        final messages = await _attachToLoggerBeforeAction(isolate2Name, action, setNewLogger: true, logLevel: RealmLogLevel.error);
+        final matches = _matchedMessagesCountPerLevel(messages, generatedName);
+        sendPort.send(matches[RealmLogLevel.error]!);
+      }, isolate2ReceivePort.sendPort);
+
+      final rootMessages = await rootIsolate;
+      final rootMatches = _matchedMessagesCountPerLevel(rootMessages, generatedName);
+      results.addEntries([MapEntry(rootIsolateName, rootMatches[RealmLogLevel.error]!)]);
+
+      results.addEntries([MapEntry(isolate1Name, await isolate1ReceivePort.first as int)]);
+      results.addEntries([MapEntry(isolate2Name, await isolate2ReceivePort.first as int)]);
+
+      isolate1ReceivePort.close();
+      (await isolate1).kill(priority: Isolate.immediate);
```

Comment:
Why do we need to do this?

Reply:
why don't we use Isolate.exit() in each isolate. Also why don't we use `Isolate.run` instead?

Reply:
I kill them after both isolates complete, because on this way I have 2 errors logged in both isolates, even though they log by one error.